modules/default/weatherforecast/weatherforecast.js

```diff
@@ -408,9 +408,11 @@ Module.register("weatherforecast", {
 		}
 
 		//Log.log(this.forecast);
-		this.show(this.config.animationSpeed, { lockString: this.identifier });
-		this.loaded = true;
-		this.updateDom(this.config.animationSpeed);
+		if (!this.hidden) {
```

Comment:
in currentweather is working but in weatherforecast is not 
only with `if (this.hidden) {`

Reply:
Hm... Is it working without the "!"? Maybe I did something wrong... But if it works now. Thanks! :)

Reply:
wierd thing, in my full version on windows 10 not showing forecast but on my ipad webpage version without node.js is working

Reply:
I do not get it to work without the "!" on the if, are you sure?

Reply:
maybe to leave another line outside of if
this.show(this.config.animationSpeed, { lockString: this.identifier });

Reply:
yes, on my windows configuration not working. on web only version is working with !

Reply:
Strange, for me it does not work if I don't include the `this.show(this.config.animationSpeed, { lockString: this.identifier });` line inside the if... I'm running the code on a "ubuntu server" and using a Windows 10 with Chrome to view it...

Reply:
OK, for me it's work with with this
```
		if (!this.hidden) {
			this.show(this.config.animationSpeed, { lockString: this.identifier });
			this.loaded = true;
			this.updateDom(this.config.animationSpeed);
		} else this.show(this.config.animationSpeed, { lockString: this.identifier });
```

Reply:
anyway, in devtools console show
```
11:07:29.242 module.js:201 weatherforecast is suspended.
11:07:34.607 module.js:208 weatherforecast is resumed.
```
nothing about currentweather that is updated according to updateInterval

Reply:
> OK, for me it's work with this
> 
> ```
> 		if (!this.hidden) {
> 			this.show(this.config.animationSpeed, { lockString: this.identifier });
> 			this.loaded = true;
> 			this.updateDom(this.config.animationSpeed);
> 		}  // this line is for compatibility, maybe in my configuration is something wrong
> 		else this.show(this.config.animationSpeed, { lockString: this.identifier });
> ```

It's not working for me with the "else-line" module shows up automatically again. 
What happens for you if you don't have the "else-line"?
I have a "test" rig that I use, only have the clock, MMM-Remote-Control, currentweather, watheerforcat and MMM-Modulebar modules running. Then I hide the "weatherforcast" module with a button on the modulebar. I have set the weatherforcast to update every minute, so after waiting one minute I see if it appears again... Maybe may testing is flawed?

Reply:
without "else" and with "!this-hiden" not working, 
without "esle" and with "this-hiden" is working,
with "esle" and with "!this-hiden" is working,

Reply:
Very strange indeed. For me the only one working is the "fist" (without else and with "!this-hidden"). The other two gives the same as it before the change, module reappears when it updates (when it's hidden without a lock). But the "code" works on all three options for me at least. Is the behavior different depending on other things? I'm running the "mirror" on "Ubuntu server 18.04.5 LTS"... And "viewing" it on a Windows 10 with chrome...

Reply:
`this.show(this.config.animationSpeed, { lockString: this.identifier });`
this line is inside and outside of if

Reply:
Yes, like below, does not work, modules shows up when it updates.
`                //Log.log(this.forecast);

                if (!this.hidden) {
                        this.show(this.config.animationSpeed, { lockString: this.identifier });
                        this.loaded = true;
                        this.updateDom(this.config.animationSpeed);
                }  // this line is for compatibility, maybe in my configuration is something wrong
                else this.show(this.config.animationSpeed, { lockString: this.identifier });
`
or like this: 
`                //Log.log(this.forecast);

                if (this.hidden) {
                        this.show(this.config.animationSpeed, { lockString: this.identifier });
                        this.loaded = true;
                        this.updateDom(this.config.animationSpeed);
                }  
`
This below works, module stays hidden even if it updates.

`                //Log.log(this.forecast);

                 if (!this.hidden) {
                      this.show(this.config.animationSpeed, { lockString: this.identifier });
                      this.loaded = true;
                      this.updateDom(this.config.animationSpeed);
              }
`
For me... :)

Reply:
OK, leave it as is goes, I will make changes that works for me.
Thank you anyway!

Reply:
Great! Thank you! :)